scripts/test_mail-alarm.py

```diff
@@ -0,0 +1,224 @@
+    &lt;alarm_trigger_period value=&quot;60&quot;/&gt;
+    &lt;alarm_auto_inhibit_period value=&quot;300&quot;/&gt;
+&lt;/variable&gt;
+'''
+
+def mock_setup(module):
+    # Emulate functions with Mock
+    module.log_err = log_err
+    module.get_pool_name = mock.Mock(return_value="test_pool")
+    module.get_sr_name_by_uuid = mock.Mock(return_value="test_sr")
+    module.get_VM_params = mock.Mock(return_value={"name_label":"test_VM"})
+    module.get_host_params = mock.Mock(return_value={"name_label":"test_host"})
+    module.mail_language_pack_path = './scripts/mail-languages'
+    module.branding.PRODUCT_BRAND = "XenServer"
+    module.branding.BRAND_CONSOLE = "XenCenter"
+
+def get_alarm_xml(xmlalarm_str, xmlcls_str, xmlname_str, xmlbody_str):
+    return XML_MESSAGE_TEMPLATE.format(alarm=xmlalarm_str, cls=xmlcls_str, body=xmlbody_str.format(name=xmlname_str))
+
+def log_err(err):
+    global log_file_global
+    with open(log_file_global, 'a+') as fileh:
+        fileh.write("%s: %s\n" % (sys.argv[0], err))
+
+class TestXapiMessage(unittest.TestCase):
+    def setUp(self):
+        global log_file_global
+        try:
+            self.work_dir = tempfile.mkdtemp(prefix='test-mail-alarm-')
+            log_file_global = os.path.join(self.work_dir, 'user.log')
+            src_file = './scripts/mail-alarm'
+            dst_file = os.path.join(self.work_dir, 'mailalarm.py')
+            shutil.copyfile(src_file, dst_file)
+            sys.path.append(self.work_dir)
+        except:
+            raise
+
+    def tearDown(self):
+            shutil.rmtree(self.work_dir, ignore_errors=True)
+
+    @nottest
+    def common_test_good_input(self, xmlalarm_str, xmlcls_str, xmlname_str, subject_str, body_str, xmlbody_str=XML_BODY_COMMON):
+        import mailalarm
+
+        # Emulate functions with Mock
+        mock_setup(mailalarm)
+
+        tst_xml = get_alarm_xml(xmlalarm_str, xmlcls_str, xmlname_str, xmlbody_str)
+        obj_XapiMessage = mailalarm.XapiMessage(tst_xml, 'en-US')
+        self.assertIsInstance(obj_XapiMessage, mailalarm.XapiMessage)
+
+        mail_subject = obj_XapiMessage.generate_email_subject()
+        mail_body = obj_XapiMessage.generate_email_body()
+
+        self.assertIn(subject_str, mail_subject)
+        self.assertIn(body_str, mail_body)
+
+    @nottest
+    def common_test_bad_input(self, xmlalarm_str, xmlcls_str, xmlname_str, title_str, subtitle_str, xmlbody_str=XML_BODY_COMMON):
+        global log_file_global
+        import mailalarm
+
+        # Emulate functions with Mock
+        mock_setup(mailalarm)
+
+        tst_xml = get_alarm_xml(xmlalarm_str, xmlcls_str, xmlname_str, xmlbody_str)
+        obj_XapiMessage = mailalarm.XapiMessage(tst_xml, 'en-USzzz')
+        self.assertIsInstance(obj_XapiMessage, mailalarm.XapiMessage)
+
+        mail_subject = obj_XapiMessage.generate_email_subject()
+        mail_body = obj_XapiMessage.generate_email_body()
+
+        with open(log_file_global, 'r') as fileh:
```

Comment:
A small comment. For future reference, the use of log files as done here is a bit non-standard. The ideal way would have been to follow the example in `nose` documentation, mock log_err with a `print` and then mock the redirection of the output to a string. Anyways, this is a good start and we are going to merge the tests as they are, keeping in mind that there is a small improvement that could be done in the future.